Feature/valve handling

Delta252 opened the pull request:
Moved valve handling from software to firmware. Confirmed to work on hardware.
Maybe this should be tested on the software as well before merging?

There are now two modes to valve handling:
[sID1000 rID100x PK1 Vy] : Opens valve y, closes valves less than y, sets valves greater than y to middle position.
[sID1000 rID100x PK1 Uy] : Where y is a 5 long digit string consisting of only digits 0-2, sets valve position according to the digit in the string. ex. 01200 - valve 1 = open, valve 2 = closed, valve 3 = middle, valve 4 = open, valve 5 = open.

When flagging an error in OpenMultipleValves(), the operation still goes through, it simply flags that one of the valves has an invalid position. This should be fine (theoretically).

Delta252 committed:
First attempt at changing valve functionality as agreed.

Created ValveHandler class

Delta252 committed:
Can now open one valve and close other valves with one command.

Tested to work on the output module.
Pending confirmation of specific behaviour.

Delta252 committed:
Modified openValveX, see lines 72-77 for functionality

Delta252 committed:
Expanded parsing of valve module.

If packet size is 2, then extract all valve states into an array.

Delta252 committed:
Now able to open multiple valves at once (Not needed currently, but might be in the future)

Added setPositionOfValves function
Changed return type of getValveStates

Delta252 committed:
Removed debugging messages

Delta252 committed:
Changed how valve handling works, should be much neater.

Removed openValvex commands and closeAllValves due to being unneeded.
Separated one valve opening case (V) and multiple valve opening case (U)
Commented functions

svirisov reviewed (changes requested):
Looks good. 
Requires example commands for all modules (valve specifically in the scope of this PR) to implement in UI.

Delta252 committed:
Standardised command parsing

Created readme with example commands
Fixed formatting
Now able to choose mixer direction

Delta252 commented:
Created readme.md in modules folder with example of all commands.

Delta252 committed:
Changed packet size values to match software

svirisov reviewed (approved):
There are some discrepancies with the UI version of expected packets:
- Syringe pump command should be [sID rID PK3 Y S{syringe type} m{volume}] for consistency with peristaltic pumps
- Sensor command not currently implemented
Otherwise, looks good.
